spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1072,17 +1076,13 @@ def _create_client_for_kernel(self, connection_file, hostname, sshkey,
                               hostname=hostname,
                               is_external_kernel=is_external_kernel,
                               is_spyder_kernel=known_spyder_kernel,
-                              slave=True,
                               show_elapsed_time=show_elapsed_time,
                               reset_warning=reset_warning,
                               ask_before_restart=ask_before_restart,
                               css_path=self.css_path,
                               configuration=self.CONFIGURATION,
-                              handlers=self.registered_spyder_kernel_handlers)
-
-        # Change stderr_dir if requested
-        if self._test_dir:
-            client.stderr_dir = self._test_dir
```

Comment:
Please revert these changes because they are not necessary for this PR, I'd say.